analyzer/src/main/kotlin/managers/Carthage.kt

```diff
@@ -0,0 +1,258 @@
+                } else {
+                    val content = URL(id).readText()
+                    val binarySpec = jsonMapper.readValue<Map<String, String>>(content)
+                    createPackageFromBinarySpec(binarySpec, id, revision)
+                }
+            }
+        }
+    }
+
+    private fun createPackageFromGenericGitUrl(projectUrl: String, revision: String): Package {
+        val vcsInfoFromUrl = VcsHost.toVcsInfo(projectUrl)
+        val vcsInfo = VcsInfo(
+            type = vcsInfoFromUrl.type,
+            url = vcsInfoFromUrl.url,
+            revision = revision
+        )
+        val vcsHost = VcsHost.toVcsHost(URI(vcsInfo.url))
+
+        return Package(
+            id = Identifier(
+                type = managerName,
+                namespace = vcsHost?.getUserOrOrganization(projectUrl).orEmpty(),
+                name = vcsHost?.getProject(projectUrl).orEmpty(),
+                version = revision
+            ),
+            declaredLicenses = sortedSetOf(),
+            description = "",
+            homepageUrl = projectUrl.removeSuffix(".git"),
+            binaryArtifact = RemoteArtifact.EMPTY,
+            sourceArtifact = RemoteArtifact.EMPTY,
+            vcs = vcsInfo
+        )
+    }
+
+    private fun createPackageFromLocalRepository(fileUrl: String, revision: String): Package {
+        val vcsInfo = VcsInfo(
+            type = VcsType.GIT,
+            url = fileUrl,
+            revision = revision
+        )
+
+        return Package(
+            id = Identifier(
+                type = managerName,
+                namespace = "",
+                name = fileUrl.substringAfterLast("/"),
+                version = revision
+            ),
+            declaredLicenses = sortedSetOf(),
+            description = "",
+            homepageUrl = "",
+            binaryArtifact = RemoteArtifact.EMPTY,
+            sourceArtifact = RemoteArtifact.EMPTY,
+            vcs = vcsInfo
+        )
+    }
+
+    private fun createPackageFromBinarySpec(binarySpec: Map<String, String>, id: String, revision: String) = Package(
+        id = Identifier(
+            type = managerName,
+            namespace = "",
+            name = id.substringAfterLast("/").removeSuffix(".json"),
+            version = revision
+        ),
+        declaredLicenses = sortedSetOf(),
+        description = "",
+        homepageUrl = "",
+
+        // The content from the artifactUrl might contain header files that should be scanned.
+        // Therefore adding this URL for both sourceArtifact and binaryArtifact.
+        binaryArtifact = RemoteArtifact(
+            url = binarySpec[revision].orEmpty(),
+            hash = Hash.NONE
+        ),
+        sourceArtifact = RemoteArtifact(
+            url = binarySpec[revision].orEmpty(),
+            hash = Hash.NONE
+        ),
+        vcs = VcsInfo.EMPTY
+    )
+
+    private fun isGithubUserAndProject(usernameProject: String) = usernameProject.split("/").size == 2
```

Comment:
IIUC there is just one caller. I believe `isGithubUserAndProject` should be inlined because:
- it's small
- only makes sense in the context of the caller
- it allows simpliciations
- it shows that the `limit = 2` in the caller is unnecessary and that `split` is done twice without need.
- I believe it makes the caller clearer.